storage-access.bs

```diff
@@ -65,53 +65,87 @@ This specification depends on the Infra standard. [[!INFRA]]
 
-Later on, Alex visits `https://video.example/`, which has an <{iframe}> on it which loads `https://social.example/heart-button-widget`. In this case, the `social.example` {{Document}} |doc| is [=third party=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
+Later on, Alex visits `https://video.example/`, which has an <{iframe}> on it which loads `https://social.example/heart-button`. In this case, the `social.example` {{Document}} |doc| is in a [=third party context=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
 
 Script in the <{iframe}> can call |doc|`.`{{Document/hasStorageAccess()}} to determine if it has access to the cookie. If it does not have access, it can request access by calling |doc|`.`{{Document/requestStorageAccess()}}.
 
 </div>
 
+<dfn>First party data</dfn> is client-side storage that would be available to a [=site=] were it loaded in a [=first party context=].
+
+To determine if a [=/browsing context=] |bc| is a <dfn>first party context</dfn>, run the following steps:
+
+1. If |bc| is a [=top-level browsing context=], return true.
+1. If |bc|'s [=parent browsing context=] is not a [=first party context=], return false.
+1. Let |settings| be the [=relevant settings object=] of |bc|'s [=active document=].
+1. If |settings|' [=environment settings object/origin=] and [=top-level origin=] are [=same site=] with one another, return true.
+1. Return false.
+
+ISSUE: This should be prose, not an algorithm.
+
+A <dfn>third party context</dfn> is a [=/browsing context=] which is not a [=first party context=].
+
 <h3 id="ua-state">User Agent state related to storage access</h3>
 
-Each User Agent maintains a <dfn>storage access map</dfn>, an [=ordered map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
+A <dfn>storage access map</dfn> is a [=map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
 
-ISSUE: What's the lifetime of the [=storage access map=] and of its entries?
+User Agents maintain a single <dfn>global storage access map</dfn>.
 
-ISSUE(28): Tie the [=storage access map=] to agent clusters.
+ISSUE(2): What is the lifecycle of the [=global storage access map=]? How long do we remember its contents? Firefox and Safari differ here.
 
-A <dfn>partitioned storage key</dfn> is a [=pair=] consisting of a top-level site and a site, both [=sites=].
+Each [=agent cluster=] has a <dfn for="agent cluster">storage access map</dfn>.
 
-ISSUE: Replace this [=partitioned storage key=] concept with whatever Anne comes up with for [[!STORAGE]].
+When an [=agent cluster=] is created, its [=agent cluster/storage access map=] is initialized with a copy of the [=global storage access map=].
```

Comment:
https://infra.spec.whatwg.org/#map-clone

Reply:
Fixed in 381972e.